core/src/main/scala/org/apache/spark/resource/TaskResourceRequest.scala

```diff
@@ -20,11 +20,15 @@ package org.apache.spark.resource
 import org.apache.spark.annotation.{Evolving, Since}
 
 /**
- * A task resource request. This is used in conjunction with the ResourceProfile to
+ * A task resource request. This is used in conjunction with the [[ResourceProfile]] to
  * programmatically specify the resources needed for an RDD that will be applied at the
  * stage level.
  *
- * Use TaskResourceRequests class as a convenience API.
+ * Use [[TaskResourceRequests]] class as a convenience API.
+ *
+ * @param resourceName Resource name
+ * @param amount Expected amount of the resource, must be either less than or equal to 0.5
```

Comment:
perhaps should clarify to be similar to text in TaskResourceRequests:

 @param amount Amount requesting as a Double to support fractional resource requests.
                 Valid values are less than or equal to 0.5 or whole numbers. This essentially
                 lets you configure X number of tasks to run on a single resource,
                 ie amount equals 0.5 translates into 2 tasks per resource address.